crates/rpc/src/server/api.rs

```diff
@@ -436,6 +436,16 @@ impl api_server::Api for RpcService {
             None
         };
 
+        let genesis = self
+            .get_genesis_header_with_retry()
+            .await
+            .context("Fetching genesis header from store");
+
+        if let Err(err) = genesis {
+            tracing::warn!(target: COMPONENT, "Failed to fetch genesis header: {}", err);
+            return Err(Status::invalid_argument("Failed to fetch genesis header".to_string()));
+        }
```

Comment:
Not a strong opinion but you can `map_err` here so that you don't have to call `unwrap()` later:


```suggestion
        let genesis = self
            .get_genesis_header_with_retry()
            .await
            .context("fetching genesis header from store")
            .map_err(|err| {
                tracing::warn!(target: COMPONENT, "Failed to fetch genesis header: {err}");
                Status::invalid_argument("failed to fetch genesis header")
            })?;
```